src/importer.js

```diff
@@ -250,28 +266,39 @@ function Importer (dagService, options) {
         })
 
         n.data = d.marshal()
+
+        pendingWrites++
         dagService.add(n, (err) => {
+          pendingWrites--
           if (err) {
             this.push({error: 'failed to store dirNode'})
+          } else if (base) {
```

Comment:
'base' is the current path of the DAG Node that's being visited as the function traverses the file hierarchy tree. On the first iteration it will be 'null' (it hasn't entered a node yet?). I've renamed this to 'path'.